packages/sdk/src/types/CcdAmount.ts

```diff
@@ -31,11 +30,34 @@ class CcdAmount {
         public readonly microCcdAmount: bigint
     ) {}
 
-    public toJSON(): string {
-        return this.microCcdAmount.toString();
+    /**
+     * Get a string representation of the CCD amount.
+     * @returns {string} The string representation.
+     */
+    public toString(): string {
+        const microCcdString = this.microCcdAmount.toString();
+        const padded = microCcdString.padStart(7, '0');
+        return `${padded.slice(0, -6)}.${padded.slice(-6)}`;
+    }
+
+    /**
+     * Get a JSON-serializable representation of the CCD amount in micro CCD.
+     * @returns {bigint} The JSON-serializable representation.
+     */
+    public toJSON(): bigint {
+        return this.microCcdAmount;
     }
 }
```

Comment:
This is a breaking change, and returning a `string` is the intended value here. It is inconsistent, but we need to keep it as is for now as otherwise deserialization will fail in a number of places.